File: src/Reverse Words in a String.cpp
#include <string>
#include <iostream>
#include <sstream>
#include <algorithm>
#include <iterator>
#include <list>
using namespace std;

class Solution {
public:
    void reverseWords(string &s) {
        istringstream iss(s);
        list<string> l;
        copy(istream_iterator<string>(iss), istream_iterator<string>(), back_insert_iterator<list<string> >(l));
        l.reverse();
        copy(l.begin(), l.end(), ostream_iterator<string>(cout, " "));
    }
};

int main()
{
    Solution so;
    string s("test the word");
    so.reverseWords(s);
    return 0;
}

File: python/Reverse Words in a String.py
class Solution:
    # @param s, a string
    # @return a string
    def reverseWords(self, s):
        lword = s.split()
        lword.reverse()
        return ' '.join(lword)

if __name__ == '__main__':
    so = Solution()
    s = "test the word"
    print(so.reverseWords(s))